src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,554 @@
+                    or unit_str in UnitsTable._derived_units
+                )
+                unit_str = remove_multiplier(unit_str) if has_multiplier else unit_str
+
+            term_power *= power
+            unit_dim = self._get_dim(unit_str, term_power)
+
+            if unit_dim != None:
+                self._add_or_update_dimension(unit_dim)
+
+    def get_dimensions_dict(self):
+        return self._dimensions
+
+    def _get_dim(self, unit_str, power):
+        if unit_str in UnitsTable._fundamental_units:
+            return {UnitsTable._fundamental_units[unit_str]: power}
+        elif unit_str in UnitsTable._derived_units:
+            self._parser(UnitsTable._derived_units[unit_str], power)
+        elif unit_str in UnitsTable._derived_units_with_conversion_factor:
+            _, unit_str = UnitsTable._derived_units_with_conversion_factor[unit_str]
+            self._parser(unit_str, power)
+        else:
+            raise ValueError("Not implemented")
+
+
+def get_si_unit_from_dim(dim_list):
+    si_unit = ""
+    dim_to_unit_map = {
+        "M": "kg",
+        "L": "m",
+        "T": "s",
+        "K": "K",
+        "A": "A",
+        "mol": "mol",
+        "cd": "cd",
+        "Angle": "radian",
+        "sr": "sr",
+        "": "",
+    }
+    for key, power in zip(UnitsTable._dimension_order.values(), dim_list):
+        unit_str = dim_to_unit_map[key]
+        spacechar = " " if len(si_unit) > 0 else ""
+        if power > 1 or power < 0:
+            si_unit += spacechar + unit_str + "^" + str(power)
+        elif power == 1:
+            si_unit += spacechar + unit_str
+    return si_unit
+
+
+class Quantity(float):
+    """This class instantiates physical quantities using their real values and
+    units.
+
+    Attributes
+    ----------
+    value: Real value
+        Value of quantity is stored as float.
+
+    unit: Unit string
+        Unit of quantity is stored as string.
+
+    Methods
+    -------
+    to(to_unit)
+        Converts to given unit string.
+
+    getDimensions(unit)
+        Extracts dimensions from unit.
+
+    isDimensionless()
+        Determines type of quantity.
+
+    Returns
+    -------
+    Quantity instance.
+
+    All the instances of this class are converted to base SI units system to have
+    consistency in all arithmetic operations.
+    """
+
+    def __new__(cls, real_value, units_string):
+        return float.__new__(cls, real_value)
+
+    def __init__(self, real_value, unit_str):
+        float.__init__(real_value)
+        self._value = self.__float__()
+        self._unit = Unit(unit_str)
+        self._dimension = Dimension(unit_str)
+
+    @property
+    def value(self):
+        return self._value
+
+    @property
+    def unit(self):
+        return self._unit.user_unit
+
+    @value.setter
+    def value(self, val):
+        self._value = val
+
+    @property
+    def dimension(self):
+        return self._dimension
+
+    @property
+    def si_value(self):
```

Comment:
Maybe it's just me but I think that this property and the next one should not be part of this interface.  The unit system should be a separate concept from the quantity.

Reply:
The `si_value` and `si_unit` is added after carefully considering  following points -

1. As per the proposal, we want to consider every scalar as SI value only.
2. We want to convert every quantity to SI before sending to fluent.
3. `pint` module also maintains SI value and SI unit for each quantity.
4. Also maintaining SI value and SI unit avoids code duplication in arithmetic operator's overloading. 
5. I think current one is the simplest way to handle all above mentioned points.

Reply:
I'm not certain I'm convinced.  I guess I just think that the unit system should be 100% detangled from the quantity type.  It's two separate concepts.  I think you can achieve point 4 still and not have the `si_unit` and `si_value` functions as public properties.

Reply:
`si_value` and `si_unit` functions are now private properties.